src/snowflake/connector/storage_client.py

```diff
@@ -185,7 +185,7 @@ def get_file_header(self, filename: str) -> FileHeader | None:
     def preprocess(self) -> None:
         meta = self.meta
         logger.debug(f"Preprocessing {meta.src_file_name}")
-
+        self.get_digest()
```

Comment:
can we test it? I remember we discussed offline that we can change the chunk size to make it easier to test

Reply:
Updated `test_large_put.py` to cover the this fix. 
I manually tested by using the updated test and removing the fix. The problem was reproduced. After adding back the fix with this PR, the problem disappeared.
I also manually tested with a file 250M on aws, azure and gcp.